### PR TITLE
Tests for metrics

### PR DESCRIPTION
**Pull request by ukff**
- add broker e2e tests for v2 metrics (operationStats)

### CODE REVIEW

File: cmd/broker/broker_suite_test.go [hunk 1]
```diff
@@ -137,6 +143,23 @@ func NewBrokerSuiteTest(t *testing.T, version ...string) *BrokerSuiteTest {
 	return NewBrokerSuiteTestWithConfig(t, cfg, version...)
 }
 
+func NewBrokerSuitTestWithMetrics(t *testing.T, version ...string) *BrokerSuiteTest {
```
Comment (piotrmiskiewicz) [inline cmd/broker/broker_suite_test.go:146]: I'd add the handler to the router in all tests, why not. In such case you don't need to create another constructor

File: cmd/broker/metrics_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	deleteReq := func(iid string) string {
+		resp := suite.CallAPI(
+			"DELETE", fmt.Sprintf(
+				"oauth/v2/service_instances/%s?accepts_incomplete=true&plan_id=361c511f-f939-4621-b228-d0fb79a1fe15&service_id=47c9dcbf-ff30-448e-ab36-d3bad66ba281", iid), ``)
+		assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+		return suite.DecodeOperationID(resp)
+	}
+
+	t.Run("AssertCorrectMetricValue", func(t *testing.T) {
+		depSuc := uuid.New().String()
```
Comment (jaroslaw-pieszka) [inline cmd/broker/metrics_test.go:63]: nit: inconsistency between `-Suc` and `-Success` postfixes. 
Could we go with only one?

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by ralikio — CHANGES_REQUESTED** [review]
Please fill description.

File: internal/process/staged_manager.go [hunk 3]
```diff
@@ -284,18 +280,24 @@ func (m *StagedManager) publishEventOnFail(operation *internal.Operation, err er
...
+
 	if operation.State == domain.Succeeded && operation.Type == internal.OperationTypeDeprovision {
 		m.publisher.Publish(context.TODO(), DeprovisioningSucceeded{
 			Operation: internal.DeprovisioningOperation{Operation: *operation},
 		})
 	}
 }
+
+func (m *StagedManager) countOperation(operation internal.Operation) {
+	m.publisher.Publish(context.TODO(), OperationCounting{
```
Comment (piotrmiskiewicz) [inline internal/process/staged_manager.go:297]: OperationCounting -> OperationFinished

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**